### ISSUE TITLE
Guidelines

### ISSUE THREAD

**Issue by patrick-nicodemus** [ocaml/ocaml.org#1992]
OCaml.org/docs/guidelines contains much that is in need of updating.
Currently the complication advice section links to an old Caml page hosted on the way back machine for examples of Makefiles. A working link is better than a broken link for sure but this seems to be outdated advice, users should be pointed towards dune.

The section on how to use Emacs is weird because it describes specific keyboard shortcuts in Emacs for recompiling and jump to next error and does not mention Merlin or Tuareg. I would think that the whole section on emacs could be removed and replaced with links to Merlin and Tuareg together with a brief overview of their functionality if the GitHub pages for those pages are insufficiently clear.

**Comment by cuihtlauac** [comment]
You are right @patrick-nicodemus. In the past months, we've focused on updating the language tutorials, but the guidelines also need to be refreshed.

Beyond Compilation and Emacs, have you spotted other documents that need urgent update?

**Comment by The-Amoghavarsha** [comment]
I would support this idea of removing text editor stuff unless we have the proper support. Also, I would like to explore the idea of making the support of bare-bones text editors (such as Emacs) possible through continuous development. 

Curious to know how the OCaml team and the text editor team are bridging the gaps or the current state of progress.
